src/axolotl/monkeypatch/attention/flash_attn_4.py

```
@@ -33,15 +36,47 @@ def _get_head_dims(model_config):
-def patch_flash_attn_4(model_config=None):
-    """Patch _lazy_imports to redirect FA2 imports to FA4 if available on supported hardware."""
+def _quack_supported():
+    """Return (ok, installed_version). ``ok`` is False only when quack-kernels is installed
+    and older than ``FA4_MIN_QUACK_VERSION``; an absent/unreadable quack is treated as ok."""
+    try:
+        from importlib.metadata import PackageNotFoundError, version
+
+        installed = version("quack-kernels")
+    except PackageNotFoundError:
+        return True, None
+    except Exception:  # pylint: disable=broad-except
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)